### ISSUE TITLE
FP16 Run

### ISSUE THREAD

**Issue by jferrer21** [NVIDIA/tacotron2#76]
@yhgon

I'm using yhgon's image, hryu/pytorch:t3-1, but I'm having errors when running FP16=True with a single V100 GPU.  Is this the correct command : **python train.py --output_directory=outdir --log_directory=logdir --hparams=fp16_run=True**  ?

**Comment by jferrer21** [comment]
@rafaelvalle

I also built the docker image "tacotron2."   

AWS Instance:
NVIDIA Volta Deep Learning AMI
Version 18.08.0
Linux/Unix, Ubuntu ubuntu 16.04 | 64-bit Amazon Machine Image (AMI) | Updated: 8/23/18

`python train.py --output_directory=outdir --log_directory=logdir --hparams=fp16_run=True`

FP16 Run: True
Dynamic Loss Scaling: True
Distributed Run: False
cuDNN Enabled: True
cuDNN Benchmark: False
/home/tacotron2/layers.py:35: UserWarning: nn.init.xavier_uniform is now deprecated in favor of nn.init.xavier_uniform_.
  self.conv.weight, gain=torch.nn.init.calculate_gain(w_init_gain))
/home/tacotron2/layers.py:15: UserWarning: nn.init.xavier_uniform is now deprecated in favor of nn.init.xavier_uniform_.
  gain=torch.nn.init.calculate_gain(w_init_gain))
FP16_Optimizer processing param group 0:
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([149, 512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024, 512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024, 256])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024, 512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024, 256])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([256, 80])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([256, 256])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096, 1536])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096, 1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([128, 1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([128, 512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1, 128])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([32, 2, 31])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([128, 32])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096, 768])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096, 1024])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([4096])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([80, 1536])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([80])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1, 1536])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([1])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 80, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([512])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([80, 512, 5])
FP16_Optimizer received torch.cuda.HalfTensor with torch.Size([80])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([80])
FP16_Optimizer received torch.cuda.FloatTensor with torch.Size([80])
Epoch: 0
train.py:219: UserWarning: invalid index of a 0-dim tensor. This will be an error in PyTorch 0.5. Use tensor.item() to convert a 0-dim tensor to a Python number
  if hparams.distributed_run else loss.data[0]
OVERFLOW! Skipping step. Attempted loss scale: 4294967296
OVERFLOW! Skipping step. Attempted loss scale: 2147483648.0
OVERFLOW! Skipping step. Attempted loss scale: 1073741824.0
OVERFLOW! Skipping step. Attempted loss scale: 536870912.0
OVERFLOW! Skipping step. Attempted loss scale: 268435456.0
Epoch: 1
OVERFLOW! Skipping step. Attempted loss scale: 134217728.0
OVERFLOW! Skipping step. Attempted loss scale: 67108864.0
OVERFLOW! Skipping step. Attempted loss scale: 33554432.0
OVERFLOW! Skipping step. Attempted loss scale: 16777216.0
OVERFLOW! Skipping step. Attempted loss scale: 8388608.0
Epoch: 2
